game.py

```diff
@@ -0,0 +1,291 @@
+
+    def set_score(self,score):
+        self.player_score = score
+    
+    def increase_score(self):
+        self.player_score += 1
+    
+    def decrease_score(self):
+        self.player_score -= 1
+    
+    def reset_score(self):
+        self.player_score = 0    
+
+#main game loop
+def run_game(display_width,display_height):
+    gameExit = False # HItting the close button (closing Window)
+    gameOver = False # Internal game over view (Hitting the walls or your own tail)
+    game = SnakeGame(display_width=display_width,display_height=display_height)
+    player = Player()
+
+    snake = Snake(start_pos=game.get_center(),image_height=50,image_width=50, icon="unicorn.png")
+
+    carrot = Carrot(icon='food.png',image_height=50,image_width=50, start_pos=(0,0))
+    carrot.set_random_location(display_width,display_height)
+
+    while not gameExit:
+        while gameOver:
+	    #sets game over screen and options to continue
+            game.game_display.fill(game.white)
+            game.message_to_screen("Game Over!", game.red, -50)
+            game.message_to_screen("Press C to play again, or Q to quit", game.black, 50)
+            pygame.display.update()
+
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    gameOver = False
+                    gameExit = True
+
+                # Maps keboard strokes to close game and redo game
+                if event.type == pygame.KEYDOWN:
+                    if event.key == pygame.K_q:
+                        gameOver = False
+                        gameExit = True
+                        exit()
+                    if event.key == pygame.K_c:
+                        run_game(display_width,display_height)
+        
+        gameExit = snake.process_move()
+
+        
+        snake.move()
+
+        gameOver = game.is_game_over(snake)
+        
+
+        #fill blue
+        game.fill_blue()
+
+        game.add_icon(carrot.icon,(carrot.x,carrot.y))
+        
+        gameOver2 = snake.add_to_tail()
+        if not gameOver:
+            gameOver = gameOver2
+        head = snake.handle_tail() #drawing snake
+
+        # defines movement of snake tail
+        game.game_display.blit(head, (snake.tail_list[-1][0], snake.tail_list[-1][1]))
+
+        #  defines adding of rainbow image to user sprite
+        for XnY in snake.tail_list[:-1]:
+            rainbow = Rainbow(icon="rainbow.png", image_height=50, image_width=50, start_pos=(0,0)).icon
+            game.game_display.blit(rainbow,(XnY[0], XnY[1]))
+
+        game.score_to_screen(str(player.player_score), game.white)
+
+        pygame.display.update()
+
+        if (game.did_collide(snake,carrot)):
+            carrot.set_random_location(display_width,display_height)
+            snake.tail_size += 1
+            player.increase_score()
+        
+        game.clock.tick(game.FPS)
+
+
+    pygame.display.quit(str())
+    pygame.quit()
+
+run_game(800,600)
```

Comment:
Sometimes, it's useful to add in run_game commands (and other such run commands) in an `if __name__ == "__main__"` loop, so that if you import functions/objects from here into another file it won't automatically start up a game.